### PR TITLE
FW-841. Improve Mercator

### PR DESCRIPTION
**Pull request by yatch**
This PR implements the items listed in FW-841.

Once these changes are merged, we need to update wiki pages on Mercator:
* https://github.com/openwsn-berkeley/mercator/wiki/Mercator-Firmware
* https://github.com/openwsn-berkeley/mercator/wiki/How-to-build-the-firmware

The code is tested with FIT/IoT-LAB (`iot-lab_M3`) M3 motes, and with OpenMote-B motes (`openmote-b-24ghz`).

### CODE REVIEW

File: bsp/boards/iot-lab_M3/uart.c [hunk 1]
```diff
@@ -95,12 +95,9 @@ void uart_clearTxInterrupts(void) {
 }
 
 void uart_setCTS(bool state) {
-    
-    if (state==0x01) {
-        USART_SendData(USART1,(uint16_t)XON);
-    } else {
-        USART_SendData(USART1,(uint16_t)XOFF);
-    }
+    // do nothing since the serial connection is bridged by a TCP
```
Comment (changtengfei) [inline bsp/boards/iot-lab_M3/uart.c]: With this changes, the M3 won't be able to work locally. Is there a problem if using the current implementation on FIT/IoT-LAB?

Comment (yatch) [reply bsp/boards/iot-lab_M3/uart.c]: XON/XOFF will be transmitted to user's terminal if I understand it correctly. If you want to use M3 locally, disabling the software flow control is one of solutions... :-/

Do we have a compilation option to disable XON/XOFF? ;-)

Comment (yatch) [reply bsp/boards/iot-lab_M3/uart.c]: Thank you for the review; I think I had a wrong idea. I confirmed that the mercator firmware works without this change. I will revert this.

Comment (changtengfei) [reply bsp/boards/iot-lab_M3/uart.c]: Thanks so much for taking the comments!

**Pull request merged**